node/subsystem/src/lib.rs

```diff
@@ -72,9 +77,8 @@ impl PartialEq for ActiveLeavesUpdate {
 	///
 	/// Instead, it means equality when `activated` and `deactivated` are considered as sets.
 	fn eq(&self, other: &Self) -> bool {
-		use std::collections::HashSet;
-		self.activated.iter().collect::<HashSet<_>>() == other.activated.iter().collect::<HashSet<_>>() &&
-			self.deactivated.iter().collect::<HashSet<_>>() == other.deactivated.iter().collect::<HashSet<_>>()
+		self.activated.len() == other.activated.len() && self.activated.iter().all(|a| other.activated.contains(a))
+			&& self.deactivated.len() == other.deactivated.len() && self.deactivated.iter().all(|a| other.deactivated.contains(a))
```

Comment:
I'd reorder the checks a bit, we can check the length on both activated and deactivated as a fast path, although this probably doesn't matter performance-wise. Once thing to note though that this check is now quadratic.

Reply:
Good point on the length check reordering. Regarding the quadratic check, while that is true, I highly assume that we will here be faster because we don't allocate and most of the time there is only one element in the update.